src/main/java/org/alliancegenome/curation_api/services/GeneExpressionAnnotationService.java

```diff
@@ -0,0 +1,135 @@
+import org.alliancegenome.curation_api.services.validation.dto.fms.GeneExpressionAnnotationFmsDTOValidator;
+import org.apache.commons.lang.StringUtils;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RequestScoped
+@JBossLog
+public class GeneExpressionAnnotationService extends BaseAnnotationCrudService<GeneExpressionAnnotation, GeneExpressionAnnotationDAO> implements BaseUpsertServiceInterface<GeneExpressionAnnotation, GeneExpressionFmsDTO> {
+
+	@Inject	ExpressionAnnotationDAO expressionAnnotationDAO;
+
+	@Inject	GeneExpressionAnnotationDAO geneExpressionAnnotationDAO;
+
+	@Inject GeneService geneService;
+
+	@Inject MmoTermService mmoTermService;
+
+	@Inject	ReferenceService referenceService;
+
+	@Inject GeneExpressionAnnotationUniqueIdHelper geneExpressionAnnotationUniqueIdHelper;
+
+	@Inject	GeneExpressionAnnotationFmsDTOValidator geneExpressionAnnotationFmsDTOValidator;
+
+	@Inject DataProviderService dataProviderService;
+
+	@Inject VocabularyTermService vocabularyTermService;
+
+	@Inject	PersonDAO personDAO;
+
+	@Inject PersonService personService;
+
+	@Override
+	protected void init() {
+		setSQLDao(geneExpressionAnnotationDAO);
+	}
+
+	@Override
+	public GeneExpressionAnnotation deprecateOrDeleteAnnotationAndNotes(Long id, Boolean throwApiError, String loadDescription, Boolean deprecate) {
+		GeneExpressionAnnotation geneExpressionAnnotation = geneExpressionAnnotationDAO.find(id);
+
+		if (geneExpressionAnnotation == null) {
+			String errorMessage = "Could not find Gene Expression Annotation with id: " + id;
+			if (throwApiError) {
+				ObjectResponse<GeneExpressionAnnotation> response = new ObjectResponse<>();
+				response.addErrorMessage("id", errorMessage);
+				throw new ApiErrorException(response);
+			}
+			log.error(errorMessage);
+			return null;
+		}
+		if (deprecate) {
+			if (!geneExpressionAnnotation.getObsolete()) {
+				geneExpressionAnnotation.setObsolete(true);
+				if (authenticatedPerson.getId() != null) {
+					geneExpressionAnnotation.setUpdatedBy(personDAO.find(authenticatedPerson.getId()));
+				} else {
+					geneExpressionAnnotation.setUpdatedBy(personService.fetchByUniqueIdOrCreate(loadDescription));
+				}
+				geneExpressionAnnotation.setDateUpdated(OffsetDateTime.now());
+				return geneExpressionAnnotationDAO.persist(geneExpressionAnnotation);
+			} else {
+				return geneExpressionAnnotation;
+			}
+		} else {
+			geneExpressionAnnotationDAO.remove(id);
+		}
+		return null;
+	}
+
+	public List<Long> getAnnotationIdsByDataProvider(BackendBulkDataProvider dataProvider) {
+		Map<String, Object> params = new HashMap<>();
+		params.put(EntityFieldConstants.DATA_PROVIDER, dataProvider.sourceOrganization);
+		if (StringUtils.equals(dataProvider.sourceOrganization, "RGD") || StringUtils.equals(dataProvider.sourceOrganization, "XB")) {
+			params.put(EntityFieldConstants.PA_SUBJECT_TAXON, dataProvider.canonicalTaxonCurie);
```

Comment:
`EntityFieldConstants.PA_SUBJECT_TAXON` is `phenotypeAnnotationSubject.taxon.curie` - not what you want here

Reply:
Fixed!

Reply:
Would be better to create EntityFieldConstants.EA_SUBJECT_TAXON here.  That way if the LinkML schema changes it is easier to find the hard-coded field names.

Reply:
Moved to EntityFieldConstants